tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -1540,8 +1540,15 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
         case Apply(receiver, method, args) =>
           val newReceiver = transformExpr(receiver)
           val newArgs = args map transformExpr
+
+          /* If the receiver is maybe a hijacked class instance, and there
+           * exists a hijacked method helper for the method, use it. Methods
+           * that do not have helpers are
+           * - Reflective proxies
+           * - Methods of RuntimeLong that are not also in java.lang.Long
+           */
           if (isMaybeHijackedClass(receiver.tpe) &&
-              !Definitions.isReflProxyName(method.name)) {
+              hijackedClassMethodToHelperName.contains(method.name)) {
```

Comment:
Yes and no. What can happen now it's that we have a `j.l.Long` on which we call e.g. `compareTo`. That is inlined, and calls `j.l.Long.compare`, which we reroute as an intrinsic to `RuntimeLong.compareTo(RuntimeLong)`, but that is not inlined. We end up with `(x: j.l.Long).compareTo(RuntimeLong)`. That method does not exist in `j.l.Long` and does not have a helper. Instead, or must be desugared straightforwardly as a method call.

Before, this did not happen because the optimizer would introduce an `asInstanceOf[RuntimeLong]` on the lhs. It doesn't do that anymore because it now considers that `j.l.Long <: RuntimeLong` (which it should have all along, but not doing it was not a problem before).